nucypher/cli/actions.py

```diff
@@ -609,26 +612,100 @@ def get_or_update_configuration(emitter, config_class, filepath: str, config_opt
         return emitter.echo(config.serialize())
 
 
-def select_worker_config_file(emitter, config_file, worker_address, provider_uri, network, federated):
+def extract_checksum_address_from_filepath(filepath, config_class=UrsulaConfiguration):
+
+    pattern = re.compile(r'''
+                         (^\w+)-
+                         (0x{1}         # Then, 0x the start of the string, exactly once
+                         [0-9a-fA-F]{40}) # Followed by exactly 40 hex chars
+                         ''',
+                         re.VERBOSE)
+    filename = os.path.basename(filepath)
+    match = pattern.match(filename)
+    if not match:
+        # Extract from default by "peeking" inside the configuration file.
+        default_name = config_class.generate_filename()
+        if filename == default_name:
+            checksum_address = config_class.peek(filepath=filepath, field='checksum_address')
+
+            ###########
+            # TODO: Cleanup and deprecate worker_address in config files, leaving only checksum_address
+            if config_class == UrsulaConfiguration:
+                federated = bool(config_class.peek(filepath=filepath, field='federated_only'))
+                if not federated:
+                    checksum_address = config_class.peek(filepath=filepath, field='worker_address')
+            ###########
 
-    config_root = abspath(dirname(config_file)) if config_file else DEFAULT_CONFIG_ROOT
-    worker_config_exists = glob.glob(UrsulaConfiguration.default_filepath(config_root=config_root))
+        else:
+            raise ValueError(f"Cannot extract checksum from filepath '{filepath}'")
+    else:
+        try:
+
+            character_name, checksum_address = match.groups()
+        except ValueError:
+            raise  # TODO: Additional handling here?
+
+    if not is_checksum_address(checksum_address):
+        raise RuntimeError(f"Invalid checksum address detected in configuration file at '{filepath}'.")
+    return checksum_address
+
+
+def select_config_file(emitter,
+                       config_class,
+                       config_root: str = None,
+                       checksum_address: str = None,
+                       ) -> str:
 
-    if not worker_config_exists:
-        emitter.message("No Ursula configurations found.  run 'nucypher ursula init' then try again.", color='red')
+    #
+    # Scrape Disk Configurations
+    #
+
+    config_root = config_root or DEFAULT_CONFIG_ROOT
+    default_config_file = glob.glob(config_class.default_filepath(config_root=config_root))
+    glob_pattern = f'{config_root}/{config_class._NAME}-0x*.{config_class._CONFIG_FILE_EXTENSION}'
```

Comment:
Feels like `._NAME` is used often enough to warrant it being a public attribute/property.

Reply:
Hmm, yeah Let's make an Issue for that.

Reply:
Filed issue #1771